cmake_modules/FindSnappy.cmake

```diff
@@ -11,57 +10,78 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+#
+# Tries to find Snappy headers and libraries.
+#
+# Usage of this module as follows:
+#
+#  find_package(Snappy)
+#
+# Variables used by this module, they can change the default behaviour and need
+# to be set before calling find_package:
+#
+#  Snappy_HOME - The root of the Snappy installation.  This is considered before
+#                standard locations. The environment variable SNAPPY_HOME is also
+#                considered if this variable is not set.
+#  Snappy_USE_STATIC_LIBS - Set to ON to force the use of the static
+#                           libraries.  Default is OFF
+#
+# Variables defined by this module:
+#
+#  SNAPPY_FOUND              System has Snappy libs/headers
+#  SNAPPY_LIBRARIES          The Snappy libraries
+#  SNAPPY_INCLUDE_DIR        The location of Snappy headers
+
+# Determine parameterized roots
+if( Snappy_HOME )
+    list( APPEND _snappy_roots ${Snappy_HOME} )
+endif()
+if( NOT "$ENV{SNAPPY_HOME}" STREQUAL "")
+    file( TO_CMAKE_PATH "$ENV{SNAPPY_HOME}" _native_path )
+    list( APPEND _snappy_roots ${_native_path} )
+endif()
 
-# - Find SNAPPY (snappy.h, libsnappy.a, libsnappy.so, and libsnappy.so.1)
-# This module defines
-#  SNAPPY_INCLUDE_DIR, directory containing headers
-#  SNAPPY_LIBS, directory containing snappy libraries
-#  SNAPPY_STATIC_LIB, path to libsnappy.a
-#  SNAPPY_FOUND, whether snappy has been found
+# Try the parameterized roots first
+find_path( SNAPPY_INCLUDE_DIR NAMES snappy.h
+    PATHS ${_snappy_roots} NO_DEFAULT_PATH
+    PATH_SUFFIXES "include" )
+find_path( SNAPPY_INCLUDE_DIR NAMES snappy.h )
 
-set(SNAPPY_SEARCH_HEADER_PATHS
-  ${THIRDPARTY_PREFIX}/include
-)
+# Support preference of static libs by adjusting CMAKE_FIND_LIBRARY_SUFFIXES.
+# Patterned after FindBoost.cmake
+if( Snappy_USE_STATIC_LIBS )
```

Comment:
We're going to want the FindXXX.cmake files to find both the static and dynamic libraries so the top level `CMakeLists.txt` can choose based on the indicated linkage of `libparquet` whether to do dynamic or static linking (or if there is only a static library, like lz4, to always use the static library), if you agree? 

What I am not sure about is situations like this:

libXXX.so
   - depends: libparquet, libsnappy, liblz4

libparquet
   - depends: libsnappy, liblz4, libthrift

Suppose you want to build `libXXX` as a dynamic library, but with its dependencies all statically linked? I suppose we can add a cmake option here which is "build static library, but do not link static dependencies", but I'm not an expert on the best practices